src/Stateless/EntryActionBehaviour.cs

```diff
@@ -31,6 +30,9 @@ public Sync(Action<Transition, object[]> action, string actionDescription) : bas
                     _action = action;
                 }
 
+                internal override string MethodName {  get { return _action.TryGetMethodName(); } }
```

Comment:
I made those methods "internal" in the abstract base class (formerly named InvocationInfo) for when it's inherited by GuardCondition (where there aren't these internal classes).  I think that in order to make MethodName public here, I would have to make it public in InvocationInfo, and then it would end up public in GuardCondition ...